Tests/UnitTests/RequestCoordinatorTests.swift

```diff
@@ -0,0 +1,249 @@
+        session.lastDataTask?.completion(Data(), response, nil)
+        
+        DispatchQueue.main.async {
+            // Assert
+            XCTAssertEqual(self.session.allDataTasks.count, 2)
+            XCTAssertTrue(self.session.didCallDataTaskWithRequest)
+            XCTAssertNotNil(self.session.lastDataTask)
+            XCTAssertEqual(self.session.lastDataTask?.originalRequest, request)
+            XCTAssertTrue(self.session.lastDataTask!.didCallResume)
+            
+            expectation.fulfill()
+        }
+        
+        self.wait(for: [expectation], timeout: 2.0)
+    }
+    
+    func testEnqueueRetriableRequest_500_retries_500_fails() {
+        // Arrange
+        let expectation = self.expectation(description: "500 -> 500 fails")
+        let url = URL(string: "https://usebutton.com")!
+        let request = URLRequest(url: url)
+        
+        // Act
+        coordinator.enqueueRetriableRequest(request: request, attempt: 0, maxRetries: 1, retryIntervalInMS: 0) { data, error in
+            
+            // Assert
+            XCTAssertNil(data)
+            XCTAssertEqual(error as? TestError, TestError.known)
+            XCTAssertEqual(self.session.allDataTasks.count, 2)
+            
+            expectation.fulfill()
+        }
+        
+        let response = HTTPURLResponse(url: url, statusCode: 500, httpVersion: nil, headerFields: nil)
+        session.lastDataTask?.completion(Data(), response, nil)
+        
+        DispatchQueue.main.async {
+            self.session.lastDataTask?.completion(nil, response, TestError.known)
+        }
+        
+        self.wait(for: [expectation], timeout: 2.0)
+    }
+    
+    func testEnqueueRetriableRequest_500_retries_200_succeeds() {
+        // Arrange
+        let expectation = self.expectation(description: "500 -> 500 succeeds")
+        let url = URL(string: "https://usebutton.com")!
+        let request = URLRequest(url: url)
+        let expectedData = Data()
+        
+        // Act
+        coordinator.enqueueRetriableRequest(request: request, attempt: 0, maxRetries: 1, retryIntervalInMS: 0) { data, error in
+            
+            // Assert
+            XCTAssertNil(error)
+            XCTAssertEqual(data, expectedData)
+            XCTAssertEqual(self.session.allDataTasks.count, 2)
+            
+            expectation.fulfill()
+        }
+        
+        let response = HTTPURLResponse(url: url, statusCode: 500, httpVersion: nil, headerFields: nil)
+        session.lastDataTask?.completion(Data(), response, nil)
+        
+        DispatchQueue.main.async {
+            let successfulResponse = HTTPURLResponse(url: url, statusCode: 200, httpVersion: nil, headerFields: nil)
+            self.session.lastDataTask?.completion(expectedData, successfulResponse, nil)
+        }
+        
+        self.wait(for: [expectation], timeout: 2.0)
+    }
+
+}
```

Comment:
I don't see any tests that verify the backoff works correctly.

Reply:
I'll add